searchlib/src/test/java/com/yahoo/searchlib/rankingexpression/evaluation/EvaluationTestCase.java

```diff
@@ -176,6 +176,28 @@ public void testBooleanEvaluation() {
         tester.assertEvaluates(true, "1 && 0 + 2");
     }
 
+    @Test
+    public void testUnpack() {
+        EvaluationTester tester = new EvaluationTester();
+        tester.assertEvaluates("tensor<float>(a{},x[16]):{foo:[" +
+                               "0,0,0,0, 0,0,0,0," +
+                               "1,1,1,1, 1,1,1,1" +
+                               "],bar:[" +
+                               "0,0,0,0, 0,0,0,1," +
+                               "1,1,1,1, 1,0,0,0]}",
+                               "unpack_bits_from_int8(tensor0, float, big)",
+                               "tensor<int8>(a{},x[2]):{foo:[0,255],bar:[1,-8]}");
+
+        tester.assertEvaluates("tensor<int8>(a{},x[16]):{foo:[" +
+                               "0,0,0,0, 0,0,0,0," +
+                               "1,1,1,1, 1,1,1,1" +
+                               "],bar:[" +
+                               "1,0,0,0, 0,0,0,0," +
+                               "0,0,0,1, 1,1,1,1]}",
+                               "unpack_bits_from_int8(tensor0, int8, little)",
+                               "tensor<int8>(a{},x[2]):{foo:[0,255],bar:[1,-8]}");
```

Comment:
I thought int8 was signed [-127,128] and not unsigned [0,255]?

Reply:
Currently the "packing" is implemented in the embedder, maybe we can add the inversion as well (https://github.com/vespa-engine/vespa/blob/master/model-integration/src/test/java/ai/vespa/embedding/ColBertEmbedderTest.java)

Reply:
yes, 255 becomes -1, fixed in test.
not sure what you want with inversion, I can easily make this:
```
input: tensor<float>(d1[6],d2[8]):[[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0], [0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 1.0], [0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 1.0], [0.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0], [1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0], [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0]]
packed: tensor<int8>(dt{},x[1]):{0:1.0, 1:5.0, 2:3.0, 3:127.0, 4:-128.0, 5:-1.0}
unpacked: tensor<float>(dt{},x[8]):{0:[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0], 1:[0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 1.0], 2:[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 1.0], 3:[0.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0], 4:[1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0], 5:[1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0]}

input: tensor<float>(d1[2],d2[16]):[[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0], [0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0]]
packed: tensor<int8>(dt{},x[2]):{0:[1.0, -128.0], 1:[5.0, 1.0]}
unpacked: tensor<float>(dt{},x[16]):{0:[0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0], 1:[0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 1.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 0.0, 1.0]}
```

Reply:
Thanks, yes, that is what I meant with inversion. That we have also `pack_into_int8` (uncertain about the name. The inspiration is from [numpy.packbits](https://numpy.org/doc/stable/reference/generated/numpy.packbits.html) and [numpy.unpackbits](https://numpy.org/doc/stable/reference/generated/numpy.unpackbits.html)